internal/bootstrap/bootstrap_provider.go

```diff
@@ -200,6 +200,18 @@ func (b *GitProviderBootstrapper) ReconcileSyncConfig(ctx context.Context, optio
 		if err != nil {
 			return err
 		}
+
+		//take the sync URL and replace the hostname with the ssh hostname
+		if gitprovider.TransportType(b.syncTransportType) == gitprovider.TransportTypeSSH && b.sshHostname != "" {
```

Comment:
How is the logic here different from what's being done on L438? And/or how can we incorporate it into this method so that we keep things contained?

Reply:
The difference is on the transport type check:

`gitprovider.TransportType(b.syncTransportType)` vs `gitprovider.TransportType(b.bootstrapTransportType)`.

The check here seems at the right place as the func is  `ReconcileSyncConfig`

Reply:
The difference (or why this is required) is because of the logic in `getCloneURL` which does an early return:

```go
	cloner, ok := repository.(gitprovider.CloneableURL); ok {
		return cloner.GetCloneURL("", transport), nil
	}
```

Given this (and as discussed in private) it would be better to change the logic there to ensure the if case is traversed, to save us from a headache in the future.